File: space-game/Data/Classes/PhysicalData.vb
﻿Imports Core.Classes
Imports Data.Interfaces

Namespace Classes
    Public Class PhysicalData
        Implements IPhysicalData

        Sub New(name As String,
                radius As Distance,
                mass As Mass,
                texture As Textures,
                type As CelestialObjectType)
            Me.Radius = radius
            Me.Name = name
            Me.Mass = mass
            Me.Texture = texture
            Me.Type = type
        End Sub

        Private Sub New ()
            ' For Mapster
        End Sub

        Public Property Name As String Implements IPhysicalData.Name
        Public Property Mass As Mass Implements IPhysicalData.Mass
        Public Property Texture As Textures Implements IPhysicalData.Texture
        Public Property Type As CelestialObjectType Implements IPhysicalData.Type
        Public Property Radius As Distance Implements IPhysicalData.Radius
    End Class
End Namespace

File: space-game/Data/Classes/PhysicalData.cs
using Core.Classes;
using System.Collections.Generic;

namespace Data.Classes
{
    public class PhysicalData : IPhysicalData
    {

        public PhysicalData(string name, Distance radius, Mass mass, List<Texture> textures, CelestialObjectType type, RingData rings)
        {
            this.Radius = radius;
            this.Name = name;
            this.Mass = mass;
            this.Textures = textures;
            this.Type = type;
            this.Rings = rings;
        }

        private PhysicalData()
        {
            // For Mapster
        }

        public string Name { get; set; }
        public Mass Mass { get; set; }
        public List<Texture> Textures { get; set; }
        public CelestialObjectType Type { get; set; }
        public RingData Rings { get; set; }
        public Distance Radius { get; set; }
    }
}
